protocols/kad/src/behaviour.rs

```diff
@@ -446,10 +445,14 @@ where
             provider_record_ttl: config.provider_record_ttl,
             connection_idle_timeout: config.connection_idle_timeout,
             local_addrs: HashSet::new(),
-            caching: config.caching,
         }
     }
 
+    /// Returns the currently configured `replication_factor`.
```

Comment:
```suggestion
```

Thinking about this some more, this is only needed in tests. In those tests, we can get the value from the `Default` implementation of `KademliaConfig`. 

I don't think we should introduce a public facing method just for testing.

Reply:
fixed